flixel/input/gamepad/FlxGamepadManager.hx

```diff
@@ -401,20 +401,30 @@ class FlxGamepadManager implements IFlxInputManager
 		// and the most popular tools just turn it into a 360 controller
 
 		name = name.toLowerCase().remove("-").remove("_");
-		return if (name.contains("ouya")) OUYA; // "OUYA Game Controller"
-		else if (name.contains("wireless controller") || name.contains("ps4")) PS4; // "Wireless Controller" or "PS4 controller"
-		else if (name.contains("logitech")) LOGITECH; else if (name.contains("xbox") && name.contains("360")) XINPUT; else if (name.contains("xinput"))
-			XINPUT;
-		else if (name.contains("nintendo rvlcnt01tr"))
-			WII_REMOTE; // WiiRemote with motion plus
-		else if (name.contains("nintendo rvlcnt01"))
-			WII_REMOTE; // WiiRemote w/o  motion plus
-		else if (name.contains("mayflash wiimote pc adapter"))
-			MAYFLASH_WII_REMOTE; // WiiRemote paired to MayFlash DolphinBar (with or w/o motion plus)
-		else if (name.contains("mfi"))
-			MFI;
-		else
-			UNKNOWN;
+		return if (name.contains("ouya"))
+				OUYA; // "OUYA Game Controller"
+			else if (name.contains("wireless controller") || name.contains("ps4"))
+				PS4; // "Wireless Controller" or "PS4 controller"
+			else if (name.contains("logitech"))
+				LOGITECH;
+			else if ((name.contains("xbox") && name.contains("360")) || name.contains("xinput"))
+				XINPUT;
+			else if (name.contains("nintendo rvlcnt01tr"))
+				WII_REMOTE; // WiiRemote with motion plus
+			else if (name.contains("nintendo rvlcnt01"))
+				WII_REMOTE; // WiiRemote w/o  motion plus
+			else if (name.contains("mayflash wiimote pc adapter"))
+				MAYFLASH_WII_REMOTE; // WiiRemote paired to MayFlash DolphinBar (with or w/o motion plus)
+			else if (name.contains("mfi"))
+				MFI;
+			else if (name.contains("pro controller") || name.contains("joycon l+r"))
+				SWITCH_PRO;
+			else if (name.contains("joycon (l)"))
+				SWITCH_JOYCON_LEFT;
+			else if (name.contains("joycon (r)"))
+				SWITCH_JOYCON_RIGHT;
```

Comment:
Can always wrap this in `#if !switch` if it messes with  current switch mappings. I don't have a Switch SDK so this isn't something I can look into